encrypt-config/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+cmake_minimum_required(VERSION 3.7)
+project(encrypt-config)
+set(VERSION, "0.1")
+
+file(GLOB ENCRYPT_CONFIG_FILES  "*.cpp")
+add_executable(encrypt-config "${ENCRYPT_CONFIG_FILES}")
+include_directories(../libminifi/include  ../thirdparty/cxxopts/include)
```

Comment:
If I change it to `target_include_directories`, I get an error message which says

```
Determining if the pthread_create exist failed with the following output:
Change Dir: /home/fgerlits/src/minifi/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_c8414/fast"
/usr/bin/make -f CMakeFiles/cmTC_c8414.dir/build.make CMakeFiles/cmTC_c8414.dir/build
make[1]: Entering directory '/home/fgerlits/src/minifi/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_c8414.dir/CheckSymbolExists.c.o
/usr/bin/cc    -o CMakeFiles/cmTC_c8414.dir/CheckSymbolExists.c.o   -c /home/fgerlits/src/minifi/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c
Linking C executable cmTC_c8414
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_c8414.dir/link.txt --verbose=1
/usr/bin/cc      -rdynamic CMakeFiles/cmTC_c8414.dir/CheckSymbolExists.c.o  -o cmTC_c8414 
CMakeFiles/cmTC_c8414.dir/CheckSymbolExists.c.o: In function `main':
CheckSymbolExists.c:(.text+0x1b): undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_c8414.dir/build.make:97: recipe for target 'cmTC_c8414' failed
make[1]: *** [cmTC_c8414] Error 1
make[1]: Leaving directory '/home/fgerlits/src/minifi/build/CMakeFiles/CMakeTmp'
Makefile:126: recipe for target 'cmTC_c8414/fast' failed
make: *** [cmTC_c8414/fast] Error 2

File /home/fgerlits/src/minifi/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c:
/* */
#include <pthread.h>

int main(int argc, char** argv)
{
  (void)argv;
#ifndef pthread_create
  return ((int*)(&pthread_create))[argc];
#else
  (void)argc;
  return 0;
#endif
}
```

If I simply delete this line, then the includes are missing from the compile options.  I have tried various ways of telling CMake that `encrypt-config` depends on `minifi` and `cxxopts` and should inherit their include and link flags, but none of them worked.

I agree that the CMake build structure should be cleared up, but I suggest we do that separately from this pull request.

Reply:
As an acceptable workaround until the cleanup, I suggest changing to `target_include_directories` and adding `target_link_libraries(encrypt-config Threads::Threads)` against the pthread error.

Reply:
Hm... simply changing to `target_include_directories` worked now.  Not sure if I did something wrong earlier, or some changes since then have made it work.